policy-test/tests/api.rs

```diff
@@ -0,0 +1,234 @@
+                    &k8s::policy::ServerAuthorization {
+                        metadata: kube::api::ObjectMeta {
+                            namespace: Some(ns.clone()),
+                            name: Some("all-admin".to_string()),
+                            ..Default::default()
+                        },
+                        spec: k8s::policy::ServerAuthorizationSpec {
+                            server: k8s::policy::server_authorization::Server {
+                                name: Some("linkerd-admin".to_string()),
+                                selector: None,
+                            },
+                            client: k8s::policy::server_authorization::Client {
+                                unauthenticated: true,
+                                ..k8s::policy::server_authorization::Client::default()
+                            },
+                        },
+                    },
+                )
+                .await
+                .expect("serverauthorization must apply");
+        tracing::trace!(?server_authz);
+        let config = rx
+            .next()
+            .await
+            .expect("watch must not fail")
+            .expect("watch must return an updated config");
+        tracing::trace!(?config);
+        assert_eq!(
+            config.protocol,
+            Some(grpc::inbound::ProxyProtocol {
+                kind: Some(grpc::inbound::proxy_protocol::Kind::Http1(
+                    grpc::inbound::proxy_protocol::Http1::default()
+                )),
+            }),
+        );
+        assert_eq!(
+            config.authorizations.first().unwrap().labels,
+            Some(("name".to_string(), "all-admin".to_string()))
+                .into_iter()
+                .collect()
+        );
+        assert_eq!(
+            *config
+                .authorizations
+                .first()
+                .unwrap()
+                .authentication
+                .as_ref()
+                .unwrap(),
+            grpc::inbound::Authn {
+                permit: Some(grpc::inbound::authn::Permit::Unauthenticated(
+                    grpc::inbound::authn::PermitUnauthenticated {}
+                )),
+            }
+        );
+        assert_eq!(
+            config.labels,
+            Some(("name".to_string(), "linkerd-admin".to_string()))
+                .into_iter()
+                .collect()
+        );
+
+        // Delete the `Server` and ensure that the update reverts to the
+        // default.
+        kube::Api::<k8s::policy::Server>::namespaced(client.clone(), &ns)
+            .delete(&server.name(), &kube::api::DeleteParams::default())
+            .await
+            .expect("Server must be deleted");
+        let config = rx
+            .next()
+            .await
+            .expect("watch must not fail")
+            .expect("watch must return an updated config");
+        tracing::trace!(?config);
+        assert_eq!(
```

Comment:
I know we can make a reasonable assumption that if the default policy is back to `deny` that the other fields have properly reset, but it may be worth checking the protocol is also defaulted back to `Detect`.

Reply:
moved these into macro assertions